git.md

```diff
@@ -1,2 +1,139 @@
-# Git
+These practices:
+
+* are aspirational but when they can't be followed it's nice to explain why;
+* document heuristics, as detailed instructions may never cover all scenarios;
+* apply to shared commits (maybe after cleaning commits locally).
+
+## tl;dr
+
+* Shared commits should not be changed.
+* Changes in each shared commit should be closely related.
+
+## 1. Preserve shared history
+
+Shared commits should never change. Some Git commands preserve the history, but
+others don't. For example, `git commit --amend` and `git rebase --interactive`
+alter the git history; they help to overwrite  a messy commit history with
+a cleaner one; you should use them only before sharing your work. To cleanup
+shared history use instead `git revert`.  If you try to push altered history by
+accident, Git will stop you; to benefit from this protection  avoid `--force`
+(or `-f`) when pushing to shared repositories.
+
+See also:
+
+* [Cleaning _unshared_ history: Squashing all commits in a branch](https://youtu.be/08dhy3Zoob4).
+* [Cleaning _unshared_ history with `git rebase --interactive`](https://youtu.be/cMI8p1XhMzA).
+* [Cleaning _unshared_ history with `git commit --amend`](https://youtu.be/539pfVfr7OI).
+* [Cleaning shared history with `git revert`](https://youtu.be/A8Ld6iDqc3w).
+* [Git playlist](https://www.youtube.com/playlist?list=PLvgdJdJDL-AOHkwiaMvYhPKVjiD9vzZIo).
+
```

Comment:
FYI I added links to short videos about the commands we mention in this section.